routes/api-18/controllers/users.controller.js

```diff
@@ -149,39 +149,53 @@ class UsersController {
-                    }).then(() => {
-                        knex.raw(query1).then((result) => {
-                            if (result && result[0] && result[0].length > 0) {
-                                stat.total_tickets = result[0][0]['total_tickets'];
-                                stat.inside_event = result[0][0]['inside_event'];
-                                stat.ticketing = result[0][0]['ticketing'];
-                                stat.last_24h_first_entrance = result[0][0]['last_24h_first_entrance'];
-                                stat.last_1h_first_entrance = result[0][0]['last_1h_first_entrance'];
-                                stat.last_24h_entrance = result[0][0]['last_24h_entrance'];
-                                stat.last_24h_exit = result[0][0]['last_24h_exit'];
-                                stat.last_1h_entrance = result[0][0]['last_1h_entrance'];
-                                stat.last_1h_exit = result[0][0]['last_1h_exit'];
-                            }
-                            res.status(200).json({groups: groups, stats: stat});
 
-                        });
-                    });
+                let stat = {};
+
+                const groupStats = await knex.raw(query);
+                stat.groups = {};
+                for (let i in groupStats[0]) {
+                    stat.groups[groupStats[0][i]['__prototype']] = {
+                        total: groupStats[0][i].total
+                    };
+                }
+
+                // stat.total_tickets = result[0][0]['total_tickets'];
+                // stat.inside_event = result[0][0]['inside_event'];
+                // stat.ticketing = result[0][0]['ticketing'];
+                // stat.last_24h_first_entrance = result[0][0]['last_24h_first_entrance'];
+                // stat.last_1h_first_entrance = result[0][0]['last_1h_first_entrance'];
+                // stat.last_24h_entrance = result[0][0]['last_24h_entrance'];
+                const ticketStats = await knex.raw(query1);
+
+                if (ticketStats && ticketStats[0] && ticketStats[0].length > 0) {
+                    stat.total_tickets = ticketStats[0][0]['total_tickets'];
+                    stat.inside_event = ticketStats[0][0]['inside_event'];
+                    stat.ticketing = ticketStats[0][0]['ticketing'];
+                    stat.last_24h_first_entrance = ticketStats[0][0]['last_24h_first_entrance'];
+                    stat.last_1h_first_entrance = ticketStats[0][0]['last_1h_first_entrance'];
+                    stat.last_24h_entrance = ticketStats[0][0]['last_24h_entrance'];
+                    stat.last_24h_exit = ticketStats[0][0]['last_24h_exit'];
+                    stat.last_1h_entrance = ticketStats[0][0]['last_1h_entrance'];
+                    stat.last_1h_exit = ticketStats[0][0]['last_1h_exit'];
                 }
+
+                // TODO: deduplicate code with api_gate_routes
```

Comment:
@amotenko optimally this should be taken out to a model / helper function. Maybe take out the "stats" to an api endpoint.

copy-pasta was the quickest. Improve, or merge like this?

Reply:
merge first, then improve.
You have all the time in the world!
Until code freeze, that is

Reply:
:+1: